### PR TITLE
#187 Add allowCellular flag to Upload

### PR DESCRIPTION
**Pull request by lyio**
+ Add flag to the Dart interface and pass it into
the method channel when enqueuing jobs

+ Read `allowCellular` flag from method channel
arguments when queueing uploads in iOS. 
+ Add a second session and session configuration
to `URLSessionUploader` that sets `allowCellularAccess` to `false`. The sessions are 
supposed to be long lived objects, so we construct both of them only once and keep them around.
+ Add switch to the actual creation of the upload task
that respects the `allowCellular` flag of the job and
chooses the correct session to schedule the upload.

### CODE REVIEW

File: lib/src/flutter_uploader.dart
Comment (lyio) [inline lib/src/flutter_uploader.dart:75]: I need to revert this change again. @ened is the line length 80?

Comment (ened) [reply lib/src/flutter_uploader.dart:75]: Probably is. You can run `flutter format` to get the codebase into the accepted format.

**Comment by lyio** [comment]
Ok, now, finally, all formatting issues should be resolved. I'm very sorry about this. I was not aware of the formatters being
run.

**Comment by ened** [comment]
@lyio no worries, I will review the PR soon. Automated code checks are not run for first-time contributors, which is preventing you from seeing this quickly.

File: android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java
Comment (ened) [inline android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java:95]: is the `<Boolean>` type specifier required? Thought this could be inferred

File: android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java
Comment (ened) [inline android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java:115]: nit, keep this one

File: android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java
Comment (ened) [inline android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java:150]: as above

File: android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java
Comment (ened) [inline android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java:127]: `true` could be a documented constant

File: android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java
Comment (ened) [inline android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java:177]: `true` could be a documented constant

File: ios/Classes/SwiftFlutterUploaderPlugin.swift
Comment (ened) [inline ios/Classes/SwiftFlutterUploaderPlugin.swift:126]: `true` could be a documented constant (like on Android)

**Review by ened — CHANGES_REQUESTED** [review]
Looks all real good, thank you for the contribution.

I left a few style comments, nothing major there.

For the `allowCellular` flag itself, I did notice it has many default values scattered around. Even though the default is always `true`, it may be best to define the default only in Dart and throw on native side if the value is not passed along. A channel test would uncover such bad behaviour by the Dart side.

File: android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java
Comment (lyio) [reply android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java:95]: you are probably right

File: android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java
Comment (lyio) [reply android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java:115]: The whitespace changes stem from the formatting tool. I am not sure if I need to change any settings there.

File: android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java
Comment (lyio) [reply android/src/main/java/com/bluechilli/flutteruploader/MethodCallHandlerImpl.java:127]: You mean like `const ALLOW_CELLULAR = true;`?

**Comment by ened** [comment]
Looking good! If you have a moment, please add a entry to the `CHANGELOG.md` file (top section) and mark this PR as ready (it's currently a "draft"). I will merge right after.

**Comment by ened** [comment]
Thank you so much for your contribution. I will cut a pre-release shortly.

**Pull request merged**
